Removed hardcoded substitute from Evidence ScriptableObjects

ViMaSter opened the pull request:
## Summary
Removes the `AltEvidence` property of `Evidence` ScriptableObjects and instead adds a second parameter to `&SUBSTITUTE_EVIDENCE`.
Also updates every narrative script that uses `&SUBSTITUTE_EVIDENCE` to have whatever was formerly set inside the `AltEvidence` property of the respective piece of evidence as the second parameter.

## Before/after screenshots and/or animated gif
### Before
`&SUBSTITUTE_EVIDENCE:Bent_Coins`
![image](https://user-images.githubusercontent.com/1689033/150482726-7110ffa4-7d7c-4996-b8cb-fab88b3ed0ea.png)

### After
`&SUBSTITUTE_EVIDENCE:Bent_Coins,Stolen_Dinos_1`
![image](https://user-images.githubusercontent.com/1689033/150483178-e94429dd-6237-4c61-91d0-83edea3f2656.png)

## Testing instructions
<!--- What steps can be taken to manually verify the changes? -->
1. Run any scene with `&SUBSTITUTE_EVIDENCE` in them (i.e. `Inky - Test Scene` with `EvidenceAddtingTestScript.ink` set inside the `NarrativeScriptPlaylist`)
2. Verify all EditMode and PlayMode tests still pass

## Additional information
<!--- Check any relevant boxes with "x" -->
- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  - Closes #153

ViMaSter committed:
Removed AltEvidence from Evidence
Updated SUBSTITUTE_EVIDENCE to require explicit substitute evidence name
Updated narrative scripts affected by this change

Commit:
Update /docs

PhoebeMitchell commented on unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs:
```diff
@@ -263,12 +263,13 @@ private void PRESENT_EVIDENCE()
-    /// <example>&amp;SUBSTITUTE_EVIDENCE:Plumber_Invoice</example>
+    /// <param name="initialEvidenceName" validFiles="Assets/ScriptableObjects/Evidence/*.asset">Name of evidence to replace with the substitute</param>
+    /// <param name="substituteEvidenceName" validFiles="Assets/ScriptableObjects/Evidence/*.asset">Name of the substitute evidence</param>
+    /// <example>&amp;SUBSTITUTE_EVIDENCE:Plumber_Invoice,Bent_Coins_1</example>
```
I suggest changing Bent_Coins_1 to one of the non-test evidence items as the writers aren't really supposed to use the test evidence.

ViMaSter replied:
Sometime my brain goes into task mode and decides to not question what I'm actually doing. Since all the evidence we have right now only had test-items assigned I just... copied it over. Even in the docs. 🙃 

Definitely a good find; and changed it in https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/154/commits/bc43e63bd0a86cbecbaaa2d85496b7998a320743! <sub>(already a candidate for most impactful commit on GitHub this year I'm sure)</sub>

ViMaSter committed:
Change evidence inside SUBSTITUTE_EVIDENCE action example to only use non-test items

Commit:
Update /docs

ViMaSter commented:
Rebased this onto `develop`, so gonna need a re-approval. 👀

ViMaSter committed:
Reintroduced /unity-ggjj/Assets/Resources/InkDialogueScripts/TestScripts/EvidenceAddingTestScript.ink
Removed unused data from /unity-ggjj/Assets/ScriptableObjects
Addressed failing tests

ViMaSter committed:
Update path to actors and evidence for ActionDecoder documentation

ViMaSter committed:
Removed empty actor asset

ViMaSter committed:
Resolve issue where substituting for evidence was limited to evidence already present inside the court record (rather than all preloaded evidence)

Commit:
Update /docs

lyndsiWilliams reviewed (approved):
Code looks good, tests pass, and test script works as described! ✅
